### PR TITLE
fix: conditional logic build groups bug

### PR DESCRIPTION
**Pull request by victorvhs017**
## What does this PR do?

Fixes https://github.com/formbricks/internal/issues/1018

This PR contains two main improvements:

1. **Refactor survey logic utilities** - Simplified the `removeCondition` and `deleteEmptyGroups` functions while maintaining identical functionality
2. **Enhance Conditions Editor UX** - Disable "Create Group" button when it doesn't make logical sense (only one condition present)

### Backend/Utils Improvements

**File:** `apps/web/lib/surveyLogic/utils.ts`

- ✅ **Simplified `removeCondition` function**
  - Now returns `boolean` indicating success/failure
  - Uses reverse iteration to avoid index shifting issues when splicing arrays
  - More efficient early termination on recursive calls

- ✅ **Added `cleanupGroup` helper function** 
  - Consolidates all group cleanup logic (removing empty groups + flattening)
  - Eliminates code duplication between `removeCondition` and `deleteEmptyGroups`
  - Ensures proper sequence: remove empty groups first, then flatten if needed

- ✅ **Updated `deleteEmptyGroups`**
  - Now a simple wrapper around the new `cleanupGroup` function
  - Maintains backward compatibility

### Frontend/UX Improvements

**File:** `apps/web/modules/ui/components/conditions-editor/index.tsx`

- ✅ **Smart "Create Group" button state**
  - Button is now disabled when `conditions.conditions.length <= 1`
  - Prevents users from trying to group a single condition (which doesn't make logical sense)

## How should this be tested?

1. **Test Condition Removal**
   - Create a survey with multiple logic conditions
   - Remove conditions and verify groups are cleaned up properly
   - Test with nested condition groups
   - Verify no empty groups remain after removals

2. **Test Group Flattening**
   - Create nested condition groups
   - Remove conditions until a group has only one nested group
   - Verify the structure flattens correctly

### Manual Testing - Conditions Editor UI

1. **Test Single Condition Scenario**
   - Navigate to survey logic editor
   - Create a logic rule with only one condition
   - Click the dropdown menu (⋯) next to the condition
   - ✅ **Verify:** "Create Group" button should be **disabled/grayed out**

2. **Test Multiple Conditions Scenario**
   - Add a second condition to the logic rule
   - Click the dropdown menu (⋯) next to any condition
   - ✅ **Verify:** "Create Group" button should be **enabled and clickable**
   - Click "Create Group" and verify it works as expected

3. **Test Dynamic State Changes**
   - Start with multiple conditions (button enabled)
   - Remove conditions until only one remains
   - ✅ **Verify:** Button becomes disabled dynamically
   - Add another condition
   - ✅ **Verify:** Button becomes enabled again

### Edge Cases to Test

1. **Deeply Nested Groups**
   - Create multiple levels of nested condition groups
   - Remove conditions from various levels
   - Verify cleanup happens correctly at all levels

2. **Complex Group Structures**
   - Mix single conditions with nested groups
   - Remove conditions and verify proper flattening
   - Test with different connector types (AND/OR)

### Required

- [x] Filled out the "How to test" section in this PR
- [x] Read [How we Code at Formbricks](<[https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md](https://formbricks.com/docs/contributing/how-we-code)>)
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand bits
- [x] Ran `pnpm build`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues
- [x] First PR at Formbricks? [Please sign the CLA!](https://cla-assistant.io/formbricks/formbricks) Without it we wont be able to merge it 🙏

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Formbricks Docs if changes were necessary

### CODE REVIEW

File: apps/web/modules/ui/components/conditions-editor/index.test.tsx [hunk 1]
```diff
@@ -233,12 +233,31 @@ describe("ConditionsEditor", () => {
...
+  });
+
+  test("disables the 'Create Group' button when there's only one condition", () => {
     render(<ConditionsEditor conditions={singleCondition} config={mockConfig} callbacks={mockCallbacks} />);
     const createGroupButton = screen.getByText("environments.surveys.edit.create_group");
-    await user.click(createGroupButton);
-    expect(mockCallbacks.onCreateGroup).toHaveBeenCalledWith("cond1");
+    expect(createGroupButton).toBeDisabled();
+  });
+
```
[the remark this answers is not in the document]
Comment (victorvhs017) [reply apps/web/modules/ui/components/conditions-editor/index.test.tsx:251]: The matchers are working already.

File: apps/web/lib/surveyLogic/utils.ts [hunk 2]
```diff
@@ -94,21 +94,47 @@ export const toggleGroupConnector = (group: TConditionGroup, resourceId: string)
...
+      // Remove if empty after cleanup
+      if (condition.conditions.length === 0) {
+        group.conditions.splice(i, 1);
+      }
+    }
+  }
+
+  // Flatten if group has only one condition and it's a condition group
+  if (group.conditions.length === 1 && isConditionGroup(group.conditions[0])) {
+    group.conditions = group.conditions[0].conditions;
```
Comment (gupta-piyush19) [inline apps/web/lib/surveyLogic/utils.ts:133]: At this point, we just shift the conditions from the only child group to its parent, but we lose the operator.

image

If I remove a condition `a` here, the operator OR between the condition `b` and `c` changes to AND
image

**Review by gupta-piyush19 — CHANGES_REQUESTED** [review]
Thanks for the PR, @victorvhs017 🚀 
I've just added one comment. Could you please address that?

File: apps/web/lib/surveyLogic/utils.ts (hunk 2 above)
Comment (victorvhs017) [reply apps/web/lib/surveyLogic/utils.ts:133]: Nice catch! I'll fix that!

**Review by gupta-piyush19 — APPROVED** [review]
Thanks for the changes, @victorvhs017!  
LGTM 🚀
